src/plugins/discover/public/application/main/components/layout/discover_layout.tsx

```diff
@@ -152,9 +152,13 @@ export function DiscoverLayout({
     [filterManager, dataView, dataViews, trackUiMetric, capabilities]
   );
 
-  const onFieldEdited = useCallback(() => {
+  const onFieldEdited = useCallback(async () => {
+    if (!dataView.isPersisted()) {
+      const updateDataView = await updateAdHocDataViewId(dataView);
+      stateContainer.setAppState({ index: updateDataView.id! });
```

Comment:
I think we should replace the url state in this case, so the back button doesn't break
```suggestion
      stateContainer.replaceUrlAppState({ index: updateDataView.id! });
```

Reply:
Thanks for catch